### PR TITLE
CVSL-295: Endpoint to update the COM allocated to an offender who has a licence in flight

### PR DESCRIPTION
**Pull request by smcveigh941**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/UpdateResponsibleComRequest.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  @field:NotNull
+  val staffIdentifier: Long,
+
+  @Schema(description = "The Delius username for the responsible COM", example = "jbloggs")
+  @field:NotNull
+  val staffUsername: String,
+
+  @Schema(description = "The email address of the responsible COM", example = "jbloggs@probation.gov.uk")
+  @field:NotNull
+  val staffEmail: String,
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/UpdateResponsibleComRequest.kt:19]: Is it worth adding a telephone number here? Or leave until we see the need?

Comment (smcveigh941) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/UpdateResponsibleComRequest.kt:19]: Was just adding the details needed right now. Email is needed for notify to work, but we will have reconsiderations when we create the COM table I'm sure

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/OffenderService.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.UpdateResponsibleComRequest
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+
+@Service
+class OffenderService(
+  private val licenceRepository: LicenceRepository,
+) {
+
+  fun updateOffenderWithResponsibleCom(crn: String, newComDetails: UpdateResponsibleComRequest) {
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/OffenderService.kt:13]: Should probably be @Transactional?  Maybe add a TODO here to record an audit event associated with this licence?

**Review by timharrison-moj — COMMENTED** [review]
LGTM. Minor comment only.

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**